app/renderer/components/messageBox.js

```diff
@@ -0,0 +1,138 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const React = require('react')
+const ImmutableComponent = require('../../../js/components/immutableComponent')
+const Dialog = require('../../../js/components/dialog')
+const Button = require('../../../js/components/button')
+const SwitchControl = require('../../../js/components/switchControl')
+const appActions = require('../../../js/actions/appActions')
+const {StyleSheet, css} = require('aphrodite')
+const commonStyles = require('./styles/commonStyles')
+
+class MessageBox extends ImmutableComponent {
+  constructor () {
+    super()
+    this.onClick = this.onClick.bind(this)
+    this.onSuppressChanged = this.onSuppressChanged.bind(this)
+  }
+
+  get tabId () {
+    return this.props.tabId || ''
+  }
+
+  get title () {
+    const title = (this.props.detail && this.props.detail.get('title')) || ''
+    return title.replace(/^An embedded page at /, '')
+  }
+
+  get message () {
+    return (this.props.detail && this.props.detail.get('message')) || ''
+  }
+
+  get buttons () {
+    return (this.props.detail && this.props.detail.get('buttons')) || ['OK']
+  }
+
+  get cancelId () {
+    return this.props.detail && this.props.detail.get('cancelId')
+  }
+
+  get suppress () {
+    return (this.props.detail && this.props.detail.get('suppress')) || false
+  }
+
+  get showSuppress () {
+    return (this.props.detail && this.props.detail.get('showSuppress')) || false
+  }
+
+  onClick (e) {
+    e.stopPropagation()
+  }
+
+  onSuppressChanged (e) {
+    const detail = this.props.detail.toJS()
+    detail.suppress = !detail.suppress
+    appActions.updateMessageBox(this.tabId, detail)
+  }
+
+  onDismiss (tabId, buttonId) {
+    const response = {
+      suppress: this.suppress,
+      result: true
+    }
+
+    if (this.cancelId) {
+      response.result = buttonId !== this.cancelId
+    }
+
+    appActions.closeMessageBox(this.tabId, response)
+  }
+
+  get messageBoxButtons () {
+    const buttons = []
+
+    for (let index = (this.buttons.size - 1); index > -1; index--) {
+      buttons.push(<Button l10nId={this.buttons.get(index)}
+        className={index === 0 ? 'primaryButton' : 'whiteButton'}
+        onClick={this.onDismiss.bind(this, this.tabId, index)} />)
+    }
+
+    return buttons
```

Comment:
I did originally implement it like that- but the buttons have to be shown in reverse order (in order to keep the same "interface" as the electron `dialog.showMessageBox`).

ex: buttons would be ['OK', 'Cancel'] with cancel being index 1 (and the caller specifies cancelId === 1)